File: 2021/11/11/check_straight_line.py
"""
https://leetcode.com/problems/check-if-it-is-a-straight-line/
Author: Andrew Jarombek
Date: 11/11/2021
"""

from typing import List


def check_straight_line(coordinates: List[List[int]]) -> bool:
    for i in range(2, len(coordinates)):
        numerator1 = coordinates[i][1] - coordinates[i - 1][1]
        denominator1 = coordinates[i][0] - coordinates[i - 1][0]

        numerator2 = coordinates[i - 1][1] - coordinates[i - 2][1]
        denominator2 = coordinates[i - 1][0] - coordinates[i - 2][0]

        if (numerator1 * denominator2) != (denominator1 * numerator2):
            return False

    return True

File: 2022/10/10/checkStraightLine.ts
/**
 * https://leetcode.com/problems/check-if-it-is-a-straight-line/
 * @author Andrew Jarombek
 * @since 10/10/2022
 */

function checkStraightLine(coordinates: number[][]): boolean {
    let i = 2;

    while (i < coordinates.length) {
        const x = (coordinates[i-1][1] - coordinates[i-2][1]) * (coordinates[i][0] - coordinates[i-1][0]);
        const y = (coordinates[i-1][0] - coordinates[i-2][0]) * (coordinates[i][1] - coordinates[i-1][1]);

        if (x != y) {
            return false;
        }

        i++;
    }

    return true;
}